frontend/public/components/nav.jsx

```diff
@@ -285,21 +285,25 @@ const imagestreamsStartsWith = ['imagestreams', 'imagestreamtags'];
 const monitoringAlertsStartsWith = ['monitoring/alerts', 'monitoring/alertrules'];
 const clusterSettingsStartsWith = ['settings/cluster', 'config.openshift.io'];
 
-const MonitoringNavSection_ = ({ urls }) => {
+const MonitoringNavSection_ = ({ flags, urls }) => {
   const prometheusURL = urls[MonitoringRoutes.Prometheus];
   const grafanaURL = urls[MonitoringRoutes.Grafana];
   const kibanaURL = urls[MonitoringRoutes.Kibana];
-  return window.SERVER_FLAGS.prometheusBaseURL || window.SERVER_FLAGS.alertManagerBaseURL || prometheusURL || grafanaURL || kibanaURL
+  const showAlerts = flags[FLAGS.CAN_GET_NS] && window.SERVER_FLAGS.prometheusBaseURL;
+  const showSilences = flags[FLAGS.CAN_GET_NS] && window.SERVER_FLAGS.alertManagerBaseURL;
+  return showAlerts || showSilences || prometheusURL || grafanaURL || kibanaURL
     ? <NavSection title="Monitoring">
-      {window.SERVER_FLAGS.prometheusBaseURL && <HrefLink href="/monitoring/alerts" name="Alerts" startsWith={monitoringAlertsStartsWith} />}
-      {window.SERVER_FLAGS.alertManagerBaseURL && <HrefLink href="/monitoring/silences" name="Silences" />}
+      {showAlerts && <HrefLink href="/monitoring/alerts" name="Alerts" startsWith={monitoringAlertsStartsWith} />}
+      {showSilences && <HrefLink href="/monitoring/silences" name="Silences" />}
       {prometheusURL && <ExternalLink href={prometheusURL} name="Metrics" />}
       {grafanaURL && <ExternalLink href={grafanaURL} name="Dashboards" />}
```

Comment:
Not new, but I wonder if we should hide prometheus and grafana links if the user can't get namespaces? Presumably they wouldn't be able to use either link due to RBAC.

Reply:
Yes, my understanding is that you need the same RBAC access for the Prometheus and Grafana UIs, so I have updated the PR to hide those links too.

I did not change the Kibana link.